src/parse_fc.c

```diff
@@ -268,18 +306,51 @@ struct policy_node *parse_fc_file(const char *filename, const struct string_list
 			continue;
 		}
 
-		struct fc_entry *entry = parse_fc_line(line);
+		if (is_within_ifdef) {
+			conditional = malloc(sizeof(struct conditional_data));
+			conditional->flavor = CONDITION_IFDEF;
+			conditional->condition = strdup(ifdef_condition);
+		} else if (is_within_ifndef) {
+			conditional = malloc(sizeof(struct conditional_data));
+			conditional->flavor = CONDITION_IFNDEF;
+			conditional->condition = strdup(ifndef_condition);
+		}
+
+		struct fc_entry *entry = parse_fc_line(line, conditional);
+		free(conditional);
```

Comment:
Given that you're freeing it immediately after the call, why not allocate on the stack rather than the heap?

Reply:
On second thought, you allocate a copy of it in parse_fc_line().  Why not allocate it here and then take ownership of that copy in parse_fc_line?